src/controllers/user.ts

```diff
@@ -0,0 +1,90 @@
+import {checkoutUserData, deleteUserData, getUserById, updateUserData} from "../services/user";
+import {Request, Response} from "express";
+import {returnUnauthorized} from "../utils/return-unauthorized";
+import {hideDeleteStatus} from "../utils/hide-delete-status";
+
+export const ID_HEADER_NAME = "x-user-id"
+
+export const getUserHandler = (req: Request, res: Response) => {
+  const id = req.get(ID_HEADER_NAME)
+  const user = id && getUserById(id);
+
+
+  if (!user) {
+    returnUnauthorized(res)
+    return;
+  }
+
+  res
+      .status(200)
+      .send({data: hideDeleteStatus(user), error: null});
+}
+
+export const updateUserHandler = (req: Request, res: Response) => {
+  const id = req.get(ID_HEADER_NAME)
+
+  const user = id && getUserById(id);
+
+
+  if (!user) {
+    returnUnauthorized(res)
+    return;
+  }
+
+  const result = id && updateUserData(req.body, id)
```

Comment:
Validation should be either a middleware (recommended) or on the controller layer, not in service. Validation always happens as high as possible.